### ISSUE TITLE
AnnotationLocationCheck: annotation on top-level variable not flagged in compact source files (JEP 512)

### ISSUE THREAD

**Issue by vivek-0509** [checkstyle/checkstyle#20219]
Parent Issue #19971

I have read check documentation: https://checkstyle.org/checks/annotation/annotationlocation.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > config.xml <<'EOF'
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="AnnotationLocation"/>
  </module>
</module>
EOF

vivek@Viveks-MacBook-Air checkstyle % cat > InputAnnotationLocationCompact.java <<'EOF'
@SuppressWarnings("unused") int x = 0;
@Deprecated String name = "hello";
void main() { }
EOF

vivek@Viveks-MacBook-Air checkstyle % RUN_LOCALE="-Duser.language=en -Duser.country=US"
java $RUN_LOCALE -jar target/checkstyle-13.5.0-SNAPSHOT-all.jar -c config.xml InputAnnotationLocationCompact.java

Starting audit...
Audit done.
```

Same code inside an ordinary class:

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > OrdinaryAnnotationLocation.java <<'EOF'
public class OrdinaryAnnotationLocation {
    @SuppressWarnings("unused") int x = 0;
    @Deprecated String name = "hello";
}
EOF

vivek@Viveks-MacBook-Air checkstyle % java $RUN_LOCALE -jar target/checkstyle-13.5.0-SNAPSHOT-all.jar -c config.xml OrdinaryAnnotationLocation.java

Starting audit...
[ERROR] OrdinaryAnnotationLocation.java:2:5: Annotation 'SuppressWarnings' should be alone on line. [AnnotationLocation]
Audit done.
Checkstyle ends with 1 errors.
```

(Only `@SuppressWarnings("unused")` is flagged because the default `allowSamelineSingleParameterlessAnnotation=true` allows single parameterless annotations like `@Deprecated` on the same line.)

---

**Describe what you expect in detail.**

AnnotationLocationCheck must flag annotations placed on the same line as a top-level variable in a JEP 512 compact source file, exactly as it does for class fields. The check's `visitToken` skips a `VARIABLE_DEF` whose parent is not `OBJBLOCK`. Top-level variables in compact source files are children of the implicit class, so their parent is not `OBJBLOCK` and the check is bypassed entirely.

Expected on the compact input: 1 violation `Annotation 'SuppressWarnings' should be alone on line.` on `x`, identical to the ordinary class above.

**Comment by vivek-0509** [comment]
To other contributors: please do not pick this issue for now to avoid duplicate work, as I may already be working on it. There are many other open issues available, so please consider picking those and help us improve the project. Thanks for understanding!

**Comment by romani** [comment]
fix is merged
